Sming/Components/rboot/component.mk

```diff
@@ -162,24 +165,30 @@ ifndef RBOOT_EMULATION
 # $1 -> application target
 # $2 -> linker script
 # $3 -> ROM address variable (not value!)
+# $4 -> ROM size variable (not value!)
 define GenerateLinkerScriptTargets
 # Mark linker script out-of-date if ROM address differs from previous run
 -include $2.config
 ifneq ($$(GEN_$3),$$($3))
 .PHONY: $2
 endif
+ifneq ($$(GEN_$4),$$($4))
+.PHONY: $2
+endif
 # Generate linker script from template
 $2: $(RBOOT_LD_TEMPLATE)
 	$$(info LDGEN $$@)
 	$$(Q) $(AWK) 'match($$$$0, /^.*irom0_0_seg[ \t]*:[ \t]*/) { \
-			printf "%sorg = 0x40200010 + ($$($3) & 0xFFFFF), len = 1M - ($$($3) & 0xFFFFF) - 0x10\n", substr($$$$0, RSTART, RLENGTH); next \
+			printf "%sorg = 0x40200010 + ($$($3) & 0xFFFFF), len = ($$($4) & 0xFFFFF) - 0x10\n", substr($$$$0, RSTART, RLENGTH); next \
```

Comment:
Since `$4` is a size, I'd crop at 0x100000 instead of masking by 0xFFFFF.

It would be even better if the build system ensures that only reasonable sizes can be used and no modification is necessary in the linker script.